onnxruntime/core/providers/cpu/tensor/upsample.h

```diff
@@ -307,7 +307,24 @@ class UpsampleBase {
                                      const std::vector<int64_t>& input_dims,
                                      std::vector<float>& scales) const {
     for (size_t i = 0, end = input_dims.size(); i < end; ++i) {
-      scales[i] = static_cast<float>(output_dims[i]) / static_cast<float>(input_dims[i]);
+      // Handle corner case to avoid dividing by zero in the next step
+      if (input_dims[i] == 0) {
+        // Enforce that output_dim is 0, given that we cannot scale 0 by any factor to
+        // result in any non-zero value
+        ORT_ENFORCE(output_dims[i] == 0,
+                    "Input dim is zero but "
```

Comment:
Prefer returning Status for things run as part of Compute unless it's a logic error in the code that should never happen. 

nit: does the message need to be spread over so many lines? also mildly helpful to output the value of 'i' so which dimension was bad is known.

Reply:
Addressed the nit and adding the dimension to the message. This method doesn't return a status as such. Should we change the signature just to accommodate this ? As such, I think ORT_ENFORCE is not bad in this place. Please let me know if you feel differently.